### PR TITLE
feat: Add expression array join function

### PR DESCRIPTION
**Pull request by peternewman**
To go with the split one requested in #2559

### CODE REVIEW

File: shared-lib/lib/Expression/ExpressionFunctions.js [hunk 1]
```diff
@@ -40,6 +40,9 @@ export const ExpressionFunctions = {
 	split: (str, separator) => {
 		return (str + '').split(separator)
 	},
+	join: (arr = [], separator = ',') => {
+		return (arr.constructor === Array ? arr : [arr]).join(separator)
```
Comment (Julusian) [inline shared-lib/lib/Expression/ExpressionFunctions.js:44]: I think this could use `Array.isArray(arr)` to check if it is an array

**Review by Julusian — APPROVED** [review]

File: shared-lib/lib/Expression/ExpressionFunctions.js (hunk 1 above)
Comment (peternewman) [reply shared-lib/lib/Expression/ExpressionFunctions.js:44]: This sounded like it thought the equality was faster, but I'm happy to switch for readability if you'd prefer?
https://stackoverflow.com/a/26633883

Comment (Julusian) [reply shared-lib/lib/Expression/ExpressionFunctions.js:44]: I'm not sure that we should be worried about speed here, these expression executor is going to be rather suboptimal by design (its an interpreter running inside an interpreter/jit)
But I don't feel that strongly about how this should be done. you thought this was better, which is enough for me

Comment (peternewman) [reply shared-lib/lib/Expression/ExpressionFunctions.js:44]: Heh very true. Maybe readability should have won out, feel free to just push a change. There didn't seem to be much else in it compatibility wise etc.

**Pull request merged**